src/lib/templateResource.ts

```diff
@@ -0,0 +1,40 @@
+type ResourceReducer =
+  | ((resource: string, config: Config) => string)
+  | ((resource: string) => string);
```

Comment:
I tried this originally, however it would then mean the signature of the the reducers which do accept `config` would need the optional question mark and then a type guard within the function body to prevent the ` Object is possibly 'undefined'` error.

```
Type '(resource: string, config: Config) => string' is not assignable to type 'ResourceReducer'.
  Types of parameters 'config' and 'config' are incompatible.
    Type 'Config | undefined' is not assignable to type 'Config'.
      Type 'undefined' is not assignable to type 'Config'.
```

Therefore I think what we have currently is cleaner.